src/org/exist/protocolhandler/embedded/EmbeddedDownloadThread.java

```diff
@@ -70,41 +74,50 @@ public EmbeddedDownloadThread(XmldbURL url, OutputStream bos) {
      * @param url Document location in database.
      * @param bos Stream to which the document is written.
      */
-    public EmbeddedDownloadThread(BrokerPool brokerPool, XmldbURL url, OutputStream bos) {
-        xmldbURL = url;
+    public EmbeddedDownloadThread(final BrokerPool brokerPool, final XmldbURL url, final OutputStream bos) {
+        super("EmbeddedDownloadThread-" + threadInitNumber.getAndIncrement());
+        this.xmldbURL = url;
         this.bos = bos;
-        this.brokerPool=brokerPool;
-        
+
         try {
-            if (brokerPool == null)
-                brokerPool = BrokerPool.getInstance(url.getInstanceName());
+            if (brokerPool == null) {
+                this.brokerPool = BrokerPool.getInstance(url.getInstanceName());
+            } else {
+                this.brokerPool = brokerPool;
+            }
             
-            subject = brokerPool.getActiveBroker().getCurrentSubject();
-        } catch (Throwable e) {
-            //e.printStackTrace();
+            this.subject = this.brokerPool.getActiveBroker().getCurrentSubject();
```

Comment:
I think you didn't get the point - there are a lot of different code styles around